fea-rs/src/compile/glyphsapp_syntax_ext.rs

```diff
@@ -0,0 +1,222 @@
+            // new operator: resolve any existing higher priority ops on the stack
+            typed::GlyphsAppExprItem::Operator(op) => {
+                let op = Operator::from(op);
+                while op_stack.last().map(|x| x.priority()).unwrap_or_default() >= op.priority() {
+                    process_op(&mut val_stack, op_stack.pop().unwrap());
+                }
+                op_stack.push(op);
+            }
+        }
+    }
+
+    // now resolve the remaining expression
+    while let Some(op) = op_stack.pop() {
+        process_op(&mut val_stack, op);
+    }
+
+    let result = match val_stack.pop() {
+        // convert back to i16
+        Some(Value::Named(val)) => {
+            ResolvedValue::Variable(val.into_iter().map(|(k, v)| (k, v.round() as _)).collect())
+        }
+        Some(Value::Lit(val)) => ResolvedValue::Scalar(val.round() as _),
+        _ => ResolvedValue::Scalar(0),
+    };
+    assert!(val_stack.is_empty());
+    result
+}
+
+impl From<typed::GlyphsAppOperator> for Operator {
+    fn from(src: typed::GlyphsAppOperator) -> Operator {
+        match src {
+            typed::GlyphsAppOperator::Plus(_) => Operator::Plus,
+            typed::GlyphsAppOperator::Minus(_) => Operator::Minus,
+            typed::GlyphsAppOperator::Mul(_) => Operator::Mul,
+            typed::GlyphsAppOperator::Div(_) => Operator::Div,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::TokenSet;
+
+    use self::typed::AstNode;
+
+    use super::*;
+
+    fn parse_expr(text: &str) -> typed::GlyphsAppNumberExpr {
+        let node = crate::parse::parse_node(text, |p| {
+            crate::parse::grammar::expect_glyphs_number_value(p, TokenSet::EMPTY);
+        });
+        let node = typed::GlyphsAppNumber::cast(&node.into()).unwrap();
+        node.iter()
+            .find_map(typed::GlyphsAppNumberExpr::cast)
+            .unwrap()
+    }
+
+    fn simple_number_value() -> HashMap<NormalizedLocation, f64> {
+        let loc1 = NormalizedLocation::for_pos(&[("wght", 0.1)]);
+        let loc2 = NormalizedLocation::for_pos(&[("wght", 0.9)]);
+        HashMap::from([(loc1, 10.), (loc2, 23.)])
+    }
+
+    // a little helper for more readable assertions
+    fn ordered_eq<const N: usize>(lhs: &ResolvedValue, rhs: [i16; N]) -> bool {
+        let ResolvedValue::Variable(lhs) = lhs else {
+            return false;
+        };
+        let mut ordered: Vec<_> = lhs.iter().collect();
+        ordered.sort_by_key(|(k, _)| *k);
+        let ordered = ordered.iter().map(|(_, v)| **v).collect::<Vec<_>>();
+        ordered == rhs
+    }
+
+    #[test]
+    fn scalar_arithmetic() {
+        let text = "${2 + 4 * 5 / 7 - 3}";
+
+        let expr = parse_expr(text);
+        let val = resolve_glyphs_app_expr(&expr, |_| Default::default());
+        let expected = (2f64 + 4. * 5. / 7. - 3.).round() as i16;
+        assert_eq!(val, ResolvedValue::Scalar(expected));
+    }
+
```

Comment:
Missing tests: just one value, either literal or named

1. `${padding}` is valid?
1. `${42}` is valid?

Reply:
good catch, added those tests.